discovery.md

```diff
@@ -101,18 +101,295 @@ be considered an event subscriber and a target user of this API.
+in this discovery system.
+
+##### `type` entity attributes
+
+###### type
+
+- Type: `String`
+- Description: CloudEvents [`type`](https://github.com/cloudevents/spec/blob/master/spec.md#type)
+  attribute.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string, following the constraints as defined in the
+    CloudEvents spec.
+- Examples:
+  - "com.github.pull.create"
+  - "com.example.object.delete.v2"
+
+###### specversion
+
+- Type: `String` per [RFC 2046](https://tools.ietf.org/html/rfc2046)
+- Description: CloudEvents [`specversion`](https://github.com/cloudevents/spec/blob/master/spec.md#specversion)
+  that will be used for events published for this producer, event type
+  combination.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+
+###### datacontenttype
+
+- Type: `String`
+- Description: CloudEevnts [`datacontenttype`](https://github.com/cloudevents/spec/blob/master/spec.md#datacontenttype)
+  attribute. Indicating how the `data` attribute of subscribed events will be
+  encoded.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST adhere to the format specified in [RFC 2046](https://tools.ietf.org/html/rfc2046)
+
+###### dataschema
+
+- Type: `URI`
+- Description: CloudEevnts [`datacontenttype`](https://github.com/cloudevents/spec/blob/master/spec.md#dataschema)
+  attribute. This identifies the canonical storage location of the schema of
+  the `data` attribute of subscribed events.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty URI
+
+###### dataschematype
+
+- Type: `String` per [RFC 2046](https://tools.ietf.org/html/rfc2046)
+- Description: If using `dataschemacontent` for inline schema storage, the
+  `dataschematype` indicates the type of schema represented there.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST adhere to the format specified in [RFC 2046](https://tools.ietf.org/html/rfc2046)
+- Examples:
+  - "application/json"
+  -
+
+###### dataschemacontent
+
+- Type: `String`
+- Description: An inline representation of the schema of the `data` attribute
+  encoding mechanism. This is an alternative to using the `dataschema`
+  attribute.
+- Constraints:
+  - OPTIONAL
+  - If present, MUST be a non-empty string containing a schema compatible with
+    the `datacontenttype`.
+  - If `dataschama` is present, this field MUST NOT be present.
+
+##### `type` entity examples
+
+A single `type` entity.
+
+```json
+{
+  "type": "com.example.storage.object.create",
+  "specversion": "1.0",
+  "datacontenttype", "application/json",
+  "dataschema": "http://schemas.example.com/download/com.example.storage.object.create.json"
+}
+```
+
+#### `producer` entity
+
+Once discovery narrows down to a specific event `type` that we want to subscribe
+to from a specific `provider`, the appropriate `producer` entity can be
+retrieved by key. The key is a composite key of the `provider` and `type` names.
+
+##### `producer` entity Attributes
```

Comment:
Not sure if it's just me or not, but I keep mixing up Provider and Producer as I'm reading the spec. So I'm worried other people might do the same thing.  Since Producer is a term we defined in CE, I'm wondering if we should find a different term for Provider in this spec - one that doesn't start with "P".  Actually do we formally define "Provider" in here? I can't seem to find it - I thought it would be in here: https://github.com/cloudevents/spec/blob/a55fcb5874cf76448d3441d2dc7fbb9e700c7470/discovery.md#terminology

Reply:
No, I don't think we did. I suppose it was me proposing this in some comments and on Slack. _Provider_ is supposed to be a classification/grouping for source. We should definitely discuss how exactly our domain model looks like and then define the new terms.

Reply:
I believe it is even mixed up in the description @duglin was referring to. https://github.com/cloudevents/spec/blob/a55fcb5874cf76448d3441d2dc7fbb9e700c7470/discovery.md#producer-entity-attributes:

> producer entity Attributes
> The Provider entity is the main component of the discovery API. This section covers the structure of that entity and the next section describes the requirements for the query API.

The title is about `producer` but the description talks about `Provider`. It is indeed very confusing. A schematic model diagram might help.

Reply:
+1, I've been a bit confused as well. The current "provider" definition needs to be more concrete about what a provider represents, imo. Right now it's `Used in discovery for enumerating the different providers represented in this discovery system.`, which uses the term to define itself.

Reply:
+1 from me, as well.
I think the provider (do we find a more appropriate term?) will play an important role for discovery.

Reply:
Following up from the call - I updated this to `EventProvider` and make the change throughout. I also added a definition.